pkg/apis/core/v1alpha1/condition.go

```diff
@@ -17,181 +17,229 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"sort"
+
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// ConditionType type for possible conditions the resource could be in.
+// A ConditionType represents a condition a resource could be in.
 type ConditionType string
 
-// Resource conditions.
+// Condition types.
 const (
-	// Pending means that the resource create request has been received and is waiting to be fulfilled.
-	Pending ConditionType = "Pending"
-	// Creating means that the resource create request has been accepted and the resource is in
-	// the process of being created.
-	Creating ConditionType = "Creating"
-	// Deleting means that the resource is in the process of being deleted.
-	Deleting ConditionType = "Deleting"
-	// Failed means that the resource is in a failure state, for example it failed to be created.
-	Failed ConditionType = "Failed"
-	// Ready means that the resource creation has been successful and the resource is ready to
-	// accept requests and perform operations.
+	// Ready managed resources are believed to be ready to handle work.
 	Ready ConditionType = "Ready"
+
+	// Synced managed resources are believed to be in sync with the Kubernetes
+	// resources that manage their lifecycle.
+	Synced ConditionType = "Synced"
+)
+
+// A ConditionReason represents the reason a resource is in a condition.
+type ConditionReason string
+
+// Condition reasons.
+const (
+	// Available indicates the managed resource is available for use.
+	Available ConditionReason = "Managed resource is available"
```

Comment:
Do you mean the difference between `Available` and `Ready`? I don't think we've had a `Running` condition.

Think of any `Reason` as "the reason this resource  is in its current state". For example "The `Reason` this resource's `Ready` condition is status `True` because the managed resource is `Available`". In this specific example the `Reason` is redundant - the _only_ reason a resource can be ready is because it's available.

This pattern starts to make a little more sense when there's multiple reasons a condition can be in a particular status. For example "the `Reason` this resource's `Ready` condition is status `False` is because the managed resource is `Deleting`. The resource being `Unavailable`, or `Creating` are also reasons the `Ready` condition may be `False`.

Note that none of this is really of my design - I'm copying the patterns from Kubernetes Node and Pod conditions, which I presume is what Crossplane modeled its `ConditionedStatus` on.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#nodecondition-v1-core
https://kubernetes.io/docs/concepts/architecture/nodes/#condition 
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#podcondition-v1-core